.eslintrc

```diff
@@ -39,13 +39,51 @@
     ],
     "@typescript-eslint/camelcase": "off",
     "@typescript-eslint/consistent-indexed-object-style": "error",
+    "@typescript-eslint/consistent-type-assertions": "error",
     "@typescript-eslint/consistent-type-definitions": ["error", "interface"],
     "@typescript-eslint/explicit-function-return-type": "off",
     "@typescript-eslint/explicit-module-boundary-types": "off",
+    "@typescript-eslint/member-ordering": [
+      "error",
+      {
+        "default": {
+          "memberTypes": ["field", "constructor", "method", "signature"],
+          "order": "alphabetically"
+        }
+      }
+    ],
     "@typescript-eslint/naming-convention": "off",
+    "@typescript-eslint/no-base-to-string": "error",
+    "@typescript-eslint/no-confusing-non-null-assertion": "error",
+    "@typescript-eslint/no-confusing-void-expression": "error",
     "@typescript-eslint/no-duplicate-imports": "error",
+    "@typescript-eslint/no-dynamic-delete": "error",
+    "@typescript-eslint/no-extraneous-class": "error",
+    "@typescript-eslint/no-implicit-any-catch": "error",
+    "@typescript-eslint/no-invalid-void-type": "error",
+    "@typescript-eslint/non-nullable-type-assertion-style": "off",
+    "@typescript-eslint/no-parameter-properties": "error",
+    "@typescript-eslint/no-require-imports": "error",
     "@typescript-eslint/no-shadow": "error",
+    "@typescript-eslint/no-unnecessary-boolean-literal-compare": "error",
+    "@typescript-eslint/no-unnecessary-condition": "error",
+    "@typescript-eslint/no-unnecessary-qualifier": "off",
+    "@typescript-eslint/no-unnecessary-type-arguments": "off",
```

Comment:
It might be worth not allowing generic type default arguments. I find them to have very confusing behavior.  Mainly in that they don't infer if you only specify one.  So specifying one effectively specifies them all anyways

Reply:
I don't think I saw a rule with that, but I will double-check! I'm in agreement with you there.

Reply:
It doesn't look like this is a rule, and I think it's a niche enough case that we're fine not writing a custom rule for it 😅